clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1948,7 +1948,7 @@ public OffsetAndMetadata committed(TopicPartition partition, final Duration time
             Timer timer = time.timer(timeout);
             Map<String, List<PartitionInfo>> topicMetadata = fetcher.getTopicMetadata(
                     new MetadataRequest.Builder(Collections.singletonList(topic), metadata.allowAutoTopicCreation()), timer);
-            return topicMetadata.get(topic);
+            return topicMetadata.getOrDefault(topic, Collections.emptyList());
```

Comment:
just curious.

Is it possible that an existent topic has no partitions? If so, returning empty list may confuse users who querying nonexistent topic. Maybe we should throw ```UnknownTopicOrPartitionException``` for nonexistent topic and return empty list for existent topic.

Reply:
BTW, it is worth writing docs for this case (return empty list).

Reply:
I don't think you could have a topic with 0 partition.